### PR TITLE
feature: Add GcpBearerAuthCredentialProvider for Schema Registry Clients

### LINKED ISSUE #25: Add BearerAuthCredentialProvider for Schema Registry Clients
Thanks for stopping by to let us know something could be better!

 **Is your feature request related to a problem? Please describe.**
I want to be able to connect to Schema Registry and authenticate using the Application Default Credential(ADC) token provided by GCP.

 **Describe the solution you'd like**
Schema Registry provides a [BearerAuthCredentialProvider](https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/security/bearerauth/BearerAuthCredentialProvider.java) interface which can implemented by a custom class. When implementing `String getBearerToken(URL url)` we can fetch the ADC from the machine on which the client is running.

### PR DESCRIPTION
**Pull request by Dippatel98**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Follow the [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [ ] Ensure the tests and linter pass
- [x] Communicate test infrastructure changes, i.e. API enablement, secrets
- [ ] Appropriate docs were updated (if necessary)

🛠️ Fixes #25

### CODE REVIEW

**Comment by Dippatel98** [comment]
r: @acocuzzo

File: kafka-java-auth/src/main/java/com/google/cloud/hosted/kafka/auth/GcpBearerAuthCredentialProvider.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  @VisibleForTesting
+  public GcpBearerAuthCredentialProvider(GoogleCredentials credentials) {
+    this.credentials = credentials;
+  }
+
+  @Override
+  public String alias() {
+    return "GCP";
```
Comment (acocuzzo) [inline kafka-java-auth/src/main/java/com/google/cloud/hosted/kafka/auth/GcpBearerAuthCredentialProvider.java:60]: Nit: make private static final

**Review by acocuzzo — APPROVED** [review]
LGTM with small nit

File: kafka-java-auth/src/main/java/com/google/cloud/hosted/kafka/auth/GcpBearerAuthCredentialProvider.java (hunk 1 above)
Comment (Dippatel98) [reply kafka-java-auth/src/main/java/com/google/cloud/hosted/kafka/auth/GcpBearerAuthCredentialProvider.java:60]: Are you talking about the string `"GCP"`?

Comment (acocuzzo) [reply kafka-java-auth/src/main/java/com/google/cloud/hosted/kafka/auth/GcpBearerAuthCredentialProvider.java:60]: yes, exactly

**Comment by acocuzzo** [comment]
@Dippatel98 to fix conventional commits you can name the PR something like "fix: xxxx" or "feature: xxxx"
and please create an issue + add to description

File: kafka-java-auth/src/main/java/com/google/cloud/hosted/kafka/auth/GcpBearerAuthCredentialProvider.java (hunk 1 above)
Comment (Dippatel98) [reply kafka-java-auth/src/main/java/com/google/cloud/hosted/kafka/auth/GcpBearerAuthCredentialProvider.java]: Done

**Pull request merged**